### PR TITLE
Price structure

### PR DESCRIPTION
**Pull request by KatBirk**
Pricing structure functionality

### CODE REVIEW

File: app/Http/Controllers/LessonController.php [hunk 1]
```diff
@@ -75,7 +76,7 @@ public function adminDoCreate(Request $request): Application|RedirectResponse|Re
             'age_max' => 'required|integer|gte:age_min',
             'season_start' => 'required|date',
             'season_end' => 'required|date|after_or_equal:season_start',
-            'price' => 'required|numeric|min:0',
+            'pricing_structure'=>'required',
```
Comment (Darchie4) [inline app/Http/Controllers/LessonController.php:79]: Should probably check that the price structure exists in DB

File: app/Http/Controllers/LessonController.php [hunk 2]
```diff
@@ -153,7 +155,7 @@ public function adminDoEdit(Request $request, int $lessonID): RedirectResponse
             'age_max' => 'required|integer|gte:age_min',
             'season_start' => 'required|date',
             'season_end' => 'required|date|after_or_equal:season_start',
-            'price' => 'required|numeric|min:0',
+            'pricing_structure' => 'required',
```
Comment (Darchie4) [inline app/Http/Controllers/LessonController.php:158]: Same again with DB check

File: resources/views/lesson/admin/create.blade.php [hunk 3]
```diff
@@ -99,8 +100,16 @@
...
 
-                <label for="price">{{__('lesson.admin_create_price')}}</label> <a href="#">{{__('lesson.admin_create_link_priceStructure')}}</a><br>
-                <input class="form-control" id="price" name="price" type="number" required><br>
+
+                <label for="pricing_structure">{{__('lesson.admin_create_price')}}</label><a href="{{route("admin.pricing.create")}}">{{__('lesson.admin_create_link_priceStructure')}}</a><br>
+                <select class="form-control form-select" id="pricing_structure" name="pricing_structure" required>
+                    <option disabled selected>{{ __('pricing.choose')}}</option>
+                    @foreach($pricings as $pricing)
+                        <option
+                            value="{{$pricing->id}}">{{$pricing->name .' ('. $pricing->price.' '.'kr. - '}} {{__('pricing.' . $pricing->payment_frequency) . ')'}}</option>
```
Comment (Darchie4) [inline resources/views/lesson/admin/create.blade.php:109]: Should probably have a lable for currency

File: resources/views/lesson/admin/edit.blade.php [hunk 4]
```diff
@@ -93,10 +94,19 @@
...
+                <label for="pricing_structure">{{__('lesson.admin_create_price')}}</label> <a href="{{route("admin.pricing.create")}}">{{__('lesson.admin_create_link_priceStructure')}}</a><br>
+                <select class="form-control form-select" id="pricing_structure" name="pricing_structure"
+                       {{ Auth::user()->can('lessons_crud') ? '' : 'disabled' }}  required>
+                    @php
+                        $selectedPricing = old('pricing_structure_id', $lesson->pricing_structure_id ?? null);
+                    @endphp
+                    <option disabled {{is_null($selectedPricing)? 'selected':''}}>{{ __('pricing.choose')}}</option>
+                    @foreach($pricings as $pricingOption)
+                        <option
+                            value="{{$pricingOption->id}}" {{ $selectedPricing == $pricingOption->id ? 'selected' : '' }}>{{$pricingOption->name .' ('. $pricingOption->price.' '.'kr. - '}} {{__('pricing.' . $pricingOption->payment_frequency) . ')'}}</option>
```
Comment (Darchie4) [inline resources/views/lesson/admin/edit.blade.php:106]: Same again with the currency

File: resources/views/lesson/admin/show.blade.php [hunk 5]
```diff
@@ -15,7 +15,12 @@
                     <b>{{__('lesson.admin_show_danseStyle')}}</b> {{$lesson->danceStyle->name}}<br>
                     <b>{{__('lesson.admin_show_difficulty')}}</b> {{$lesson->difficulty->name}}<br>
                     <b>{{__('lesson.admin_show_ageGroup')}}</b> {{$lesson->age_min}} - {{$lesson->age_max}} år<br>
-                    <b>{{__('lesson.admin_show_price')}}</b> {{$lesson->price}} DKK.<br>
+                    <b>{{__('lesson.admin_show_price')}}</b>
+                    @if(isset($lesson->pricingStructure->name)){{$lesson->pricingStructure->name.' ('.$lesson->pricingStructure->price.' kr. - '.
```
Comment (Darchie4) [inline resources/views/lesson/admin/show.blade.php:19]: Currency lable

**Review by Darchie4 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/LessonController.php (hunk 1 above)
Comment (KatBirk) [reply app/Http/Controllers/LessonController.php:79]: Seems like unnecessary work? It is a select input generated based on the values in the db, so you can't input anything that doesn't exist

**Pull request merged**